pyfstat/helper_functions.py

```diff
@@ -327,19 +327,43 @@ def convert_array_to_gsl_matrix(array):
     return gsl_matrix
 
 
-def get_sft_array(sftfilepattern, data_duration, F0, dF0):
-    """ Return the raw data from a set of sfts """
+def get_sft_array(sftfilepattern, data_duration=None, F0=None, dF0=None):
+    """ Return the raw data (absolute value) from a set of SFTs
+
+    FIXME: currently only returns data for first detector
+    """
+
+    if F0 is None and dF0 is None:
+        fMin = -1
+        fMax = -1
+    elif F0 is None or dF0 is None:
```

Comment:
@dbkeitel  Python's `or` is the inclusive one, I think you were thinking about he exclusive one?

| F0 | dF0 | raise ValueError |
|----|-----|-----------|
| 0  | 0   | 0         | None of them: This is fine
| 1  | 0   | 1         |
| 0  | 1   | 1         |
| 1  | 1   | 0         | Both of them: This is fine!

Proposal using the Bitwise `XOR` operator
```
elif (F0 is None) ^ (dF0 is None):
```

Reply:
Since it's an `elif` to the previous case of both being `None`, I think it's valid...?

00 -> L336 catches
11 or 01 or 10 -> L339 would catch, but in 11 case it never gets that far
11 -> else L341

Reply:
Let me think about it...

0 means "None", 1 means "other than None".

###  00
336: 1 and 1 -> 0 
If none of them are given, take the whole band. Ok.

### 01, 10
336: 0 and 1 -> 0 Nothing happens
339: 0 or 1 -> 1 raise Value error
If only one of them is given, **raise error**

### 11
336: 0 and 0 -> 0 Nothing happens
339: 0 or 0 -> 0 Nothing happens
341: else -> Do something
If both are given, use them. Ok.

Reply:
Fair enough then.